app/models/ingredient_recipe.rb

```diff
@@ -0,0 +1,8 @@
+class IngredientRecipe < ActiveRecord::Base
+  belongs_to :ingredient
+  belongs_to :recipe
```

Comment:
don't repeat you self :
validates :recipe_id,:ingredient_id,:quantity,:unit, presence: true